### PR TITLE
Run precompiled vm tests for package:build

### PR DESCRIPTION
**Pull request by jakemac53**
Fix an issue with the kernel builder and package cycles, and use the new kernel pipeline to run the package:build tests with caching on travis.

### CODE REVIEW

File: build/build.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+targets:
+  $default:
+    builders:
+      build_modules|modules:
+        options:
+          strategy: coarse
```
Comment (natebosch) [inline build/build.yaml]: I'm in favor of changing this to the default

File: build_modules/lib/src/kernel_builder.dart [hunk 2]
```diff
@@ -76,39 +77,47 @@ Future _createKernel(
...
-    if (await buildStep.canRead(kernelId)) {
-      transitiveKernelDeps.add(kernelId);
-    } else {
-      transitiveSourceDeps.addAll(dep.sources);
-    }
-  }));
+  var request = new WorkRequest();
+  AssetId outputId;
+  File outputFile;
+  File packagesFile;
```
Comment (natebosch) [inline build_modules/lib/src/kernel_builder.dart]: why were these moved up?

Comment (jakemac53) [reply build_modules/lib/src/kernel_builder.dart]: They are used below but assigned inside the block so that the temporary sets within the block can be gc'd

Comment (natebosch) [reply build_modules/lib/src/kernel_builder.dart]: Ah, I missed that 84 to 152 is now a block

I think we need to split this out into some submethods since it's nearly 100 lines and the structure is getting hard to follow.

Comment (jakemac53) [reply build_modules/lib/src/kernel_builder.dart]: done
